graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -167,11 +167,60 @@ abstract class BaseDataTypeGenerator(
 ) {
     internal val typeUtils = TypeUtils(packageName, config, document)
 
+    object ParserConstants {
+        const val NAME = "name"
+        const val REASON = "reason"
+        const val CUSTOM_ANNOTATION = "annotate"
+        const val DEPRECATED = "deprecated"
+        const val REPLACE_WITH_STR = ", replace with "
+    }
+
+    /**
+     * Creates an argument map of the input Arguments
+     */
+    private fun createArgumentMap(directive: Directive): MutableMap<String, Value<Value<*>>> {
+        return directive.arguments.fold(mutableMapOf()) { argMap, argument ->
+            argMap[argument.name] = argument.value
+            argMap
+        }
+    }
+
+    /**
+     * Applies directives like customAnnotation
+     */
+    private fun applyDirectives(directives: List<Directive>): Pair<MutableList<AnnotationSpec>, String?> {
+        var commentFormat: String? = null
+        return Pair(
+            directives.fold(mutableListOf()) { annotations, directive ->
+                val argumentMap = createArgumentMap(directive)
+                if (directive.name == ParserConstants.CUSTOM_ANNOTATION && config.generateCustomAnnotations) {
+                    annotations.add(customAnnotation(argumentMap, config))
+                }
+                if (directive.name == ParserConstants.DEPRECATED) {
+                    annotations.add(deprecatedAnnotation())
+                    if (argumentMap.containsKey(ParserConstants.REASON)) {
+                        val reason: String = (argumentMap[ParserConstants.REASON] as StringValue).value
+                        val replace = reason.substringAfter(ParserConstants.REPLACE_WITH_STR, "")
+                        commentFormat = reason.substringBefore(ParserConstants.REPLACE_WITH_STR)
+                        if (replace.isNotEmpty()) {
+                            commentFormat = "@deprecated ${reason.substringBefore(ParserConstants.REPLACE_WITH_STR)}. Replaced by $replace"
+                        }
+                    } else {
+                        throw IllegalArgumentException("Deprecated requires an argument `${ParserConstants.REASON}`")
```

Comment:
I don't think java's `@Deprecated` annotation requires a `reason` as mandatory argument. 
In fact, it doesn't have it AFAIK.
A quick glance through [@Deprecated](https://docs.oracle.com/javase/9/docs/api/java/lang/Deprecated.html) looks like there are two params, `since` & `forRemoval`.

Reply:
this the check for graphql @deprecated annotation. the reason is mandatory there irrespective of java or kotlin

Reply:
Also the since and removal is not supported in the grpahql @depracated from what I have read So currently wont be able to support it